codersdk/workspacebuilds.go

```diff
@@ -65,13 +65,15 @@ type WorkspaceBuild struct {
 	InitiatorID         uuid.UUID           `json:"initiator_id" format:"uuid"`
 	InitiatorUsername   string              `json:"initiator_name"`
 	Job                 ProvisionerJob      `json:"job"`
-	Reason              BuildReason         `db:"reason" json:"reason"`
+	Reason              BuildReason         `db:"reason" json:"reason" enums:"initiator,autostart,autostop"`
```

Comment:
Nice work! Once we get the docs in shape, it'd be nice to try to process the Go code to find these enums automatically.

Btw, would it be possible to move this enum description to the type? Like:

```go
// BuildReason ...
// @enum initiator,autostart,autostop
type BuildReason string
```

Perhaps not supported by swaggo.

Reply:
There are some magic undocumented extensions like `x-enum-varnames` or `x-enum-descriptions`, but I don't think that this is what we're looking for.

See https://github.com/swaggo/swag/issues/984

Reply:
Indeed, that seems useful with enums that are number based and have string representations in (de-)serialization.